src/helpers/assertStringsWithinText.ts

```diff
@@ -0,0 +1,28 @@
+import { ReactTestRendererNode } from 'react-test-renderer';
+
+export const assertStringsWithinText = (
+  rendererJSON: ReactTestRendererNode | null | ReactTestRendererNode[]
+) => {
+  if (!rendererJSON) return;
+
+  if (Array.isArray(rendererJSON)) {
+    rendererJSON.forEach(assertStringsWithinText);
+    return;
+  }
+
+  if (typeof rendererJSON === 'string') {
+    return;
+  }
+
+  if (rendererJSON.type !== 'Text') {
+    if (rendererJSON.children?.some((child) => typeof child === 'string')) {
+      throw new Error(
+        'Text strings must be rendered within a host Text component.'
+      );
+    }
```

Comment:
What about `number` is that changed to `string` by `toJSON` method?

Reply:
Yes it is, there is a test case to ensure rendering numbers outside text components does throw